certs/factory.go

```diff
@@ -0,0 +1,86 @@
+package certs
+
+import "fmt"
+
+type certKeyPair struct {
+	cert []byte
+	key  []byte
+}
+
+var clientCertMap = map[string]certKeyPair{
+	"administrator": {
+		cert: AdministratorCert,
+		key:  AdministratorKey,
+	},
+	"badclient": {
+		cert: BadClientCert,
+		key:  BadClientKey,
+	},
+	"client1": {
+		cert: Client1Cert,
+		key:  Client1Key,
+	},
+	"client2": {
+		cert: Client2Cert,
+		key:  Client2Key,
+	},
+	"client3": {
+		cert: Client3Cert,
+		key:  Client3Key,
+	},
+	"weakclient": {
+		cert: WeakClientCert,
+		key:  WeakClientKey,
+	},
+}
+
+/*
+var serverCertMap = map[string]certKeyPair{
+	"badserver": {
+		cert: BadServerCert,
+		key:  BadServerKey,
+	},
+	"server": {
+		cert: ServerCert,
+		key:  ServerKey,
+	},
+	"weakserver": {
+		cert: WeakServerCert,
+		key:  WeakServerKey,
+	},
+}
+*/
+
+func ClientFactory(userID string) (cert, key []byte, err error) {
```

Comment:
I have a few suggestions regarding this setup:

* Place all certificate constants in a single file? Seems simpler to manage.
* Ditch the factory concept and just use the constants in pairs?

Reply:
In the yet-to-come client, I'm going to want to easily map a userID (e.g., `"client1`") to the pairs to enable me to easily test with different client-side keys.  This is here to make that easier.

Yes, I could put them all in one file, but this way, if I ever want to do away with a key/cert set, it's easier to just delete a couple of files than delete a couple of files and go edit another one.